From 6e8a1526d9592269b31d258ac971c061e3eff2f7 Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Tue, 22 Aug 2017 15:40:26 -0500
Subject: [PATCH] Prepare for our NMS patches


diff --git a/applyPatches.sh b/applyPatches.sh
index 02401133..516739df 100755
--- a/applyPatches.sh
+++ b/applyPatches.sh
@@ -8,9 +8,7 @@ fi
 
 nms=$1/net/minecraft/server
 cb=src/main/java/net/minecraft/server
-#clean up and rebuild
-rm -rf $cb
-mkdir -p $cb
+
 for file in $(/bin/ls nms-patches)
 do
     patchFile="nms-patches/$file"
@@ -19,6 +17,8 @@ do
     echo "Patching $file < $patchFile"
     sed -i 's/\r//' "$nms/$file" > /dev/null
 
+    rm "$cb/$file"
+
     cp "$nms/$file" "$cb/$file"
     patch -d src/main/java/ "net/minecraft/server/$file" < "$patchFile"
 done
diff --git a/makePatches.sh b/makePatches.sh
index 3400e7e9..00be047f 100755
--- a/makePatches.sh
+++ b/makePatches.sh
@@ -5,10 +5,19 @@ then
     echo "Please run this script again with the clean decompile sources as an argument. In most cases this will be ../work/decompile-XXXX"
     exit
 fi
+
+if [ -z "$2" ]
+then
+    echo "You need to specify the project file(s) you want to generate a patch for."
+    exit
+fi
+
 cb=src/main/java/net/minecraft/server
 nms="$1/net/minecraft/server"
 
-for file in $(/bin/ls $cb)
+shift
+
+for file in $@
 do
     echo "Diffing $file"
     sed -i 's/\r//' "$nms/$file"
-- 
2.17.0

