From 694d128324196ceedfa4c7b374a8da984229aac1 Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Tue, 27 Dec 2016 18:48:49 -0600
Subject: [PATCH] Implement hopper toggling API.


diff --git a/src/main/java/net/minecraft/server/TileEntityHopper.java b/src/main/java/net/minecraft/server/TileEntityHopper.java
index 559eedfa..2eb6b442 100644
--- a/src/main/java/net/minecraft/server/TileEntityHopper.java
+++ b/src/main/java/net/minecraft/server/TileEntityHopper.java
@@ -15,6 +15,10 @@ import org.bukkit.event.inventory.InventoryPickupItemEvent;
 import org.bukkit.inventory.Inventory;
 // CraftBukkit end
 
+// CweepahCraft start
+import org.bukkit.Bukkit;
+// CweepahCraft end
+
 public class TileEntityHopper extends TileEntityLootable implements IHopper, ITickable {
 
     private NonNullList<ItemStack> items;
@@ -109,7 +113,7 @@ public class TileEntityHopper extends TileEntityLootable implements IHopper, ITi
     }
 
     public void tick() {
-        if (this.world != null && !this.world.isClientSide) {
+        if (Bukkit.isHopperEnabled() && this.world != null && !this.world.isClientSide) {
             --this.f;
             this.j = this.world.getTime();
             if (!this.E()) {
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index a0afe056..4c139214 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -201,6 +201,7 @@ public final class CraftServer implements Server {
     public boolean ignoreVanillaPermissions = false;
     private final List<CraftPlayer> playerView;
     public int reloadCount;
+    private boolean hopperEnabled = true;
 
     private final class BooleanWrapper {
         private boolean value = true;
@@ -1845,6 +1846,16 @@ public final class CraftServer implements Server {
         return false;
     }
 
+    @Override
+    public boolean isHopperEnabled() {
+        return hopperEnabled;
+    }
+
+    @Override
+    public void setHopperEnabled(boolean enabled) {
+        hopperEnabled = enabled;
+    }
+
     @Override
     public Entity getEntity(UUID uuid) {
         Validate.notNull(uuid, "UUID cannot be null");
-- 
2.21.0

