From 2991ddab36785553c5599cf6ac3ff61d33f3da9e Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Tue, 27 Dec 2016 18:50:33 -0600
Subject: [PATCH] Prepend username to logged chat messages.


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 8cede938..687025ca 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -1332,7 +1332,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                         }
 
                         String message = String.format(queueEvent.getFormat(), queueEvent.getPlayer().getDisplayName(), queueEvent.getMessage());
-                        PlayerConnection.this.minecraftServer.console.sendMessage(message);
+                        PlayerConnection.this.minecraftServer.console.sendMessage(queueEvent.getPlayer() + ": " + message);
                         if (((LazyPlayerSet) queueEvent.getRecipients()).isLazy()) {
                             for (Object player : PlayerConnection.this.minecraftServer.getPlayerList().players) {
                                 ((EntityPlayer) player).sendMessage(CraftChatMessage.fromString(message));
@@ -1362,7 +1362,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                 }
 
                 s = String.format(event.getFormat(), event.getPlayer().getDisplayName(), event.getMessage());
-                minecraftServer.console.sendMessage(s);
+                minecraftServer.console.sendMessage(event.getPlayer().getName() + ": " + s);
                 if (((LazyPlayerSet) event.getRecipients()).isLazy()) {
                     for (Object recipient : minecraftServer.getPlayerList().players) {
                         ((EntityPlayer) recipient).sendMessage(CraftChatMessage.fromString(s));
-- 
2.17.0

