From 4d9e79892f733df57ba4ae8aa2fdcb6f38eaec6a Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Tue, 27 Dec 2016 18:52:30 -0600
Subject: [PATCH] Fix piston dupe glitch.


diff --git a/src/main/java/net/minecraft/server/BlockPiston.java b/src/main/java/net/minecraft/server/BlockPiston.java
index dbe0ff33..08a0fe35 100644
--- a/src/main/java/net/minecraft/server/BlockPiston.java
+++ b/src/main/java/net/minecraft/server/BlockPiston.java
@@ -288,14 +288,6 @@ public class BlockPiston extends BlockDirectional {
             return false;
         } else {
             List<BlockPosition> list = pistonextendschecker.getMovedBlocks();
-            List<IBlockData> list1 = Lists.newArrayList();
-
-            for (int i = 0; i < list.size(); ++i) {
-                BlockPosition blockposition2 = (BlockPosition) list.get(i);
-
-                list1.add(world.getType(blockposition2));
-            }
-
             List<BlockPosition> list2 = pistonextendschecker.getBrokenBlocks();
             int j = list.size() + list2.size();
             IBlockData[] aiblockdata = new IBlockData[j];
@@ -363,7 +355,7 @@ public class BlockPiston extends BlockDirectional {
                 blockposition3 = blockposition3.shift(enumdirection1);
                 set.remove(blockposition3);
                 world.setTypeAndData(blockposition3, (IBlockData) Blocks.MOVING_PISTON.getBlockData().set(BlockPiston.FACING, enumdirection), 68);
-                world.setTileEntity(blockposition3, BlockPistonMoving.a((IBlockData) list1.get(k), enumdirection, flag, false));
+                world.setTileEntity(blockposition3, BlockPistonMoving.a(iblockdata, enumdirection, flag, false));
                 --j;
                 aiblockdata[j] = iblockdata;
             }
-- 
2.21.0

