From bddd934265b86cf19aa03f600aa107a5bd7d70d4 Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Sun, 7 May 2017 10:29:46 -0500
Subject: [PATCH] Implement action bar support for title API


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 2d5dc217..3b5b546c 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1437,6 +1437,11 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     @Override
     public void sendTitle(String title, String subtitle, int fadeIn, int stay, int fadeOut) {
+        sendTitle(title, subtitle, null, fadeIn, stay, fadeOut);
+    }
+
+    @Override
+    public void sendTitle(String title, String subtitle, String actionBar, int fadeIn, int stay, int fadeOut) {
         PacketPlayOutTitle times = new PacketPlayOutTitle(fadeIn, stay, fadeOut);
         getHandle().playerConnection.sendPacket(times);
 
@@ -1449,6 +1454,11 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
             PacketPlayOutTitle packetSubtitle = new PacketPlayOutTitle(EnumTitleAction.SUBTITLE, CraftChatMessage.fromString(subtitle)[0]);
             getHandle().playerConnection.sendPacket(packetSubtitle);
         }
+
+        if (actionBar != null) {
+            PacketPlayOutTitle packetActionBar = new PacketPlayOutTitle(EnumTitleAction.ACTIONBAR, CraftChatMessage.fromString(actionBar)[0]);
+            getHandle().playerConnection.sendPacket(packetActionBar);
+        }
     }
 
     @Override
-- 
2.17.0

