From b71cd12f8674f5a50a72b4723ea16fceaa049e3c Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Fri, 8 Mar 2019 12:25:35 -0600
Subject: [PATCH] Implement performCommandAsChat method


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 79771e8c..acb5464a 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -1471,12 +1471,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
 
             // CraftBukkit start
             if (isSync) {
-                try {
-                    this.minecraftServer.server.playerCommandState = true;
-                    this.handleCommand(s);
-                } finally {
-                    this.minecraftServer.server.playerCommandState = false;
-                }
+                this.handleCommand0(s);
             } else if (s.isEmpty()) {
                 LOGGER.warn(this.player.getName() + " tried to send an empty message");
             } else if (getPlayer().isConversing()) {
@@ -1549,7 +1544,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
         }
 
         if (!async && s.startsWith("/")) {
-            this.handleCommand(s);
+            this.handleCommand0(s);
         } else if (this.player.getChatFlags() == EntityHuman.EnumChatVisibility.SYSTEM) {
             // Do nothing, this is coming from a plugin
         } else {
@@ -1616,7 +1611,16 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
     }
     // CraftBukkit end
 
-    private void handleCommand(String s) {
+    public void handleCommand(String command) {
+        try {
+            this.minecraftServer.server.playerCommandState = true;
+            this.handleCommand0(command);
+        } finally {
+            this.minecraftServer.server.playerCommandState = false;
+        }
+    }
+
+    private void handleCommand0(String s) {
         org.bukkit.craftbukkit.SpigotTimings.playerCommandTimer.startTiming(); // Spigot
         // CraftBukkit start - whole method
         if ( org.spigotmc.SpigotConfig.logCommands ) // Spigot
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 607f7098..2c8776e4 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -330,6 +330,11 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return server.dispatchCommand(this, command);
     }
 
+    @Override
+    public void performCommandAsChat(String command) {
+        getHandle().playerConnection.handleCommand(command);
+    }
+
     @Override
     public void playNote(Location loc, byte instrument, byte note) {
         if (getHandle().playerConnection == null) return;
-- 
2.21.0

