From beba86e2bf4559eea9af8a4273fd5c9e2486ef2b Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Sat, 10 Jun 2017 10:52:56 -0500
Subject: [PATCH] Parrot improvements


diff --git a/src/main/java/net/minecraft/server/EntityHuman.java b/src/main/java/net/minecraft/server/EntityHuman.java
index c374efd7..d76a9b40 100644
--- a/src/main/java/net/minecraft/server/EntityHuman.java
+++ b/src/main/java/net/minecraft/server/EntityHuman.java
@@ -397,7 +397,7 @@ public abstract class EntityHuman extends EntityLiving {
 
         this.j(this.getShoulderEntityLeft());
         this.j(this.getShoulderEntityRight());
-        if (!this.world.isClientSide && (this.fallDistance > 0.5F || this.isInWater() || this.isPassenger()) || this.abilities.isFlying) {
+        if (!this.world.isClientSide && (this.fallDistance > 3.0F || this.isInWater() || this.abilities.isFlying || (this.isSneaking() && !this.onGround))) {
             this.releaseShoulderEntities();
         }
 
diff --git a/src/main/java/net/minecraft/server/EntityParrot.java b/src/main/java/net/minecraft/server/EntityParrot.java
index f31d9b31..e827ab96 100644
--- a/src/main/java/net/minecraft/server/EntityParrot.java
+++ b/src/main/java/net/minecraft/server/EntityParrot.java
@@ -61,7 +61,7 @@ public class EntityParrot extends EntityPerchable implements EntityBird {
         super.initAttributes();
         this.getAttributeMap().b(GenericAttributes.e);
         this.getAttributeInstance(GenericAttributes.maxHealth).setValue(6.0D);
-        this.getAttributeInstance(GenericAttributes.e).setValue(0.4000000059604645D);
+        this.getAttributeInstance(GenericAttributes.e).setValue(4 * 0.4000000059604645D);
         this.getAttributeInstance(GenericAttributes.MOVEMENT_SPEED).setValue(0.20000000298023224D);
     }
 
diff --git a/src/main/java/net/minecraft/server/PathfinderGoalSit.java b/src/main/java/net/minecraft/server/PathfinderGoalSit.java
index d80fce98..c2bf1b4c 100644
--- a/src/main/java/net/minecraft/server/PathfinderGoalSit.java
+++ b/src/main/java/net/minecraft/server/PathfinderGoalSit.java
@@ -20,7 +20,7 @@ public class PathfinderGoalSit extends PathfinderGoal {
         } else {
             EntityLiving entityliving = this.entity.getOwner();
 
-            return entityliving == null ? true : (this.entity.h(entityliving) < 144.0D && entityliving.getLastDamager() != null ? false : this.willSit);
+            return entityliving == null ? true : (!(this.entity instanceof EntityParrot) && this.entity.h(entityliving) < 144.0D && entityliving.getLastDamager() != null ? false : this.willSit);
         }
     }
 
-- 
2.17.0

