From 09a2fceae9ecdc296d2abc7d937f6b837d2b5989 Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Fri, 16 Jun 2017 13:29:45 -0500
Subject: [PATCH] Avoid potential frozen entities from Dynmap


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 740633d9..f298f66f 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1082,6 +1082,13 @@ public abstract class World implements IBlockAccess {
             }
 
             this.getChunkAt(i, j).a(entity);
+
+            // CweepahCraft start - Remove matching entity from removal queue (if present)
+            if (this.f.remove(entity)) {
+                getServer().getLogger().warning("Removed an entity from removal queue with an ID that matches one being added to the world.");
+            }
+            // CweepahCraft end
+
             this.entityList.add(entity);
             this.b(entity);
             return true;
@@ -2558,6 +2565,13 @@ public abstract class World implements IBlockAccess {
         org.spigotmc.AsyncCatcher.catchOp( "entity world add"); // Spigot
         // CraftBukkit start
         // this.entityList.addAll(collection);
+
+        // CweepahCraft start - Remove any matching entities from the removal queue (if present)
+        if (this.f.removeAll(collection)) {
+            getServer().getLogger().warning("Removed entities from removal queue with IDs that match ones being added to the world.");
+        }
+        // CweepahCraft end
+
         Iterator iterator = collection.iterator();
 
         while (iterator.hasNext()) {
-- 
2.17.0

