From 775aa03bd333f15c86c7bf1bb3a4aedac8e823fb Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Sun, 3 Sep 2017 20:21:26 -0500
Subject: [PATCH] Fix initialization of explorer maps in plugin-created worlds


diff --git a/src/main/java/net/minecraft/server/ItemWorldMap.java b/src/main/java/net/minecraft/server/ItemWorldMap.java
index e0799ff0..676d017d 100644
--- a/src/main/java/net/minecraft/server/ItemWorldMap.java
+++ b/src/main/java/net/minecraft/server/ItemWorldMap.java
@@ -194,8 +194,10 @@ public class ItemWorldMap extends ItemWorldMapBase {
         if (itemstack.getItem() == Items.FILLED_MAP) {
             WorldMap worldmap = Items.FILLED_MAP.getSavedMap(itemstack, world);
 
-            if (worldmap != null) {
-                if (world.worldProvider.getDimensionManager().getDimensionID() == worldmap.map) {
+            // CweepahCraft start - don't assume dimension type identifier is equal to the dimension identifier
+            if (worldmap != null && world.getMinecraftServer() != null) {
+                if (world == world.getMinecraftServer().getWorldServer(worldmap.map)) {
+                    // CweepahCraft end
                     int i = 1 << worldmap.scale;
                     int j = worldmap.centerX;
                     int k = worldmap.centerZ;
-- 
2.14.1

