From 60b9f651df8cb9e518acbaee1aaf0a6d5b725a50 Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Tue, 27 Dec 2016 12:31:46 -0600
Subject: [PATCH] Hopper toggling API


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index d8014f14..8ac7fb3d 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -1191,6 +1191,14 @@ public final class Bukkit {
         return server.createBossBar(title, color, style, flags);
     }
 
+    public static boolean isHopperEnabled() {
+        return server.isHopperEnabled();
+    }
+
+    public static void setHopperEnabled(boolean enabled) {
+        server.setHopperEnabled(enabled);
+    }
+
     /**
      * Creates a boss bar instance to display to players. The progress defaults
      * to 1.0.
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index fac90a48..6df5ba95 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1048,6 +1048,10 @@ public interface Server extends PluginMessageRecipient {
      */
     boolean removeBossBar(NamespacedKey key);
 
+    boolean isHopperEnabled();
+
+    void setHopperEnabled(boolean enabled);
+
     /**
      * Gets an entity on the server by its UUID
      *
-- 
2.21.0

