From 260e4732f941e6f32ace9035d2235f6b99bb3296 Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Mon, 13 Mar 2017 15:52:08 -0500
Subject: [PATCH] Add time sync API


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index d62277de..f1121306 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -775,6 +775,27 @@ public interface World extends PluginMessageRecipient, Metadatable {
      */
     public void setFullTime(long time);
 
+    /**
+     * Ensure this world's time is synchronized with the specified world's time. Set to null to disable.
+     *
+     * @param timeSync Whether or not this world's time should be in sync with the main world.
+     */
+    public void setTimeSync(World timeSync);
+
+    /**
+     * Returns the world that this world's time is synchronized with. Null if none.
+     *
+     * @return World this world's time is synchronized with.
+     */
+    public World getTimeSync();
+
+    /**
+     * Returns whether or not this world's time is synchronized with another's.
+     *
+     * @return Whether or not this world's time is synchronized with another's.
+     */
+    public boolean hasTimeSync();
+
     /**
      * Returns whether the world has an ongoing storm.
      *
-- 
2.21.0

