From c55115cf2e727470fafc3e466bb07defd160aaa6 Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Mon, 15 May 2017 19:25:42 -0500
Subject: [PATCH] API to send chat messages as chat message type


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 29bc604d..4b6ca1d4 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1455,6 +1455,13 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
      */
     public String getLocale();
 
+    /**
+     * Sends this sender a message deemed a chat message
+     *
+     * @param message Message to be displayed
+     */
+    public void sendChatMessage(String message);
+
     // Spigot start
     public class Spigot extends Entity.Spigot
     {
-- 
2.17.0

