From 7e1029e8c3c5524c8c7843981aa7b64a1bed73db Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Mon, 15 May 2017 19:25:42 -0500
Subject: [PATCH] API to send chat messages as chat message type


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 70cde9d2..c5847dff 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1461,6 +1461,13 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     public String getLocale();
 
+    /**
+     * Sends this sender a message deemed a chat message
+     *
+     * @param message Message to be displayed
+     */
+    public void sendChatMessage(String message);
+
     /**
      * Update the list of commands sent to the client.
      * <br>
-- 
2.21.0

