From 09c22e3a7273a9ce5bc35d61e08ad9fdec8f9c9e Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Wed, 26 Jul 2017 08:46:08 -0500
Subject: [PATCH] Add whole-inventory containsAtLeast


diff --git a/src/main/java/org/bukkit/inventory/Inventory.java b/src/main/java/org/bukkit/inventory/Inventory.java
index bab561dc..856daf22 100644
--- a/src/main/java/org/bukkit/inventory/Inventory.java
+++ b/src/main/java/org/bukkit/inventory/Inventory.java
@@ -232,6 +232,18 @@ public interface Inventory extends Iterable<ItemStack> {
      */
     public boolean containsAtLeast(ItemStack item, int amount);
 
+    /**
+     * Checks if the inventory contains ItemStacks matching the given
+     * ItemStack whose amounts sum to at least the minimum amount specified.
+     *
+     * @param item the ItemStack to match against
+     * @param amount the minimum amount
+     * @param includeAll set true to include all contents, not just storage contents
+     * @return false if item is null, true if amount less than 1, true if
+     *     enough ItemStacks were found to add to the given amount
+     */
+    public boolean containsAtLeast(ItemStack item, int amount, boolean includeAll);
+
     /**
      * Returns a HashMap with all slots and ItemStacks in the inventory with
      * the given Material.
-- 
2.21.0

