From fac513e643b880c0d42e5503e4d8a085238dd67b Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Wed, 9 Aug 2017 13:38:23 -0500
Subject: [PATCH] Add performCommandAsChat method


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index ecadfc74..c42ff22f 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -132,6 +132,16 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
      */
     public boolean performCommand(String command);
 
+    /**
+     * Makes the player perform the given command. In this method, the
+     * command will be performed as though the player had sent it through
+     * chat, triggering PlayerCommandPreProcessEvent. Appropriately, it
+     * must be formatted as such; not omitting the leading "/".
+     *
+     * @param command Command to perform
+     */
+    public void performCommandAsChat(String command);
+
     /**
      * Returns if the player is in sneak mode
      *
-- 
2.14.1

