From 3166068d12f788f79f7949b63b65fa9a653e03c1 Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Sat, 19 Aug 2017 18:54:15 -0500
Subject: [PATCH] Add mysterious armor


diff --git a/src/main/java/org/bukkit/Material.java b/src/main/java/org/bukkit/Material.java
index 8648959..880e057 100644
--- a/src/main/java/org/bukkit/Material.java
+++ b/src/main/java/org/bukkit/Material.java
@@ -536,6 +536,10 @@ public enum Material {
     RECORD_10(2265, 1),
     RECORD_11(2266, 1),
     RECORD_12(2267, 1),
+    MYSTERIOUS_HELMET(31996, 1, 726),
+    MYSTERIOUS_CHESTPLATE(31997, 1, 1056),
+    MYSTERIOUS_LEGGINGS(31998, 1, 990),
+    MYSTERIOUS_BOOTS(31999, 1, 858)
     ;
 
     private final int id;
-- 
1.9.1

