From dde749da95604aabfcc429e9d63732c0b873268e Mon Sep 17 00:00:00 2001
From: Jacob Martin <jjm_223@hotmail.com>
Date: Sun, 3 Sep 2017 10:27:38 -0500
Subject: [PATCH] Add ClickType and InventoryAction API constants for the
 recipe book


diff --git a/src/main/java/org/bukkit/event/inventory/ClickType.java b/src/main/java/org/bukkit/event/inventory/ClickType.java
index a7440aac..c489f0dd 100644
--- a/src/main/java/org/bukkit/event/inventory/ClickType.java
+++ b/src/main/java/org/bukkit/event/inventory/ClickType.java
@@ -53,6 +53,10 @@ public enum ClickType {
      * Any action done with the Creative inventory open.
      */
     CREATIVE,
+    /**
+     * Manipulating the inventory via the recipe book.
+     */
+    AUTO_RECIPE,
     /**
      * A type of inventory manipulation not yet recognized by Bukkit.
      * <p>
diff --git a/src/main/java/org/bukkit/event/inventory/InventoryAction.java b/src/main/java/org/bukkit/event/inventory/InventoryAction.java
index a7bc694b..306104da 100644
--- a/src/main/java/org/bukkit/event/inventory/InventoryAction.java
+++ b/src/main/java/org/bukkit/event/inventory/InventoryAction.java
@@ -83,6 +83,10 @@ public enum InventoryAction {
      * the cursor up to {@link org.bukkit.Material#getMaxStackSize()}.
      */
     COLLECT_TO_CURSOR,
+    /**
+     * The inventory is being manipulated by the recipe book.
+     */
+    AUTO_RECIPE,
     /**
      * An unrecognized ClickType.
      */
-- 
2.17.0

